modules/puzzle/position_list.py

```diff
@@ -143,10 +143,9 @@ def ambiguous(self):
                 if (self.analysed_legals[0].evaluation.mate < 1
                         and self.analysed_legals[1].evaluation.mate < 1):
                     return True
-            if self.analysed_legals[0].evaluation.mate is None or self.analysed_legals[1].evaluation.cp is None:
-                return
-            if self.analysed_legals[1].evaluation.cp < -200:
-                return True
+            if self.analysed_legals[0].evaluation.mate is not None or self.analysed_legals[1].evaluation.cp is not None:
+                if self.analysed_legals[1].evaluation.cp < -200:
+                    return True
```

Comment:
The condition has an `and` 
```suggestion
            if (self.analysed_legals[0].evaluation.mate is not None
                and self.analysed_legals[1].evaluation.cp is not None):
                if (self.analysed_legals[1].evaluation.cp < -200):
                    return True
```